Topology/Spectral/basic.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Panagiotis Angelinos. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Panagiotis Angelinos
+-/
+
+import Mathlib.Topology.QuasiSeparated
+import Mathlib.Topology.Compactness.Compact
+import Mathlib.Topology.ContinuousOn
+import Mathlib.Topology.Sober
+
+/-!
+# Spectral Spaces
+This file defines spectral spaces. A topological space is spectral if it is sober,
+quasi-compact, the intersection of two quasi-compact opens is quasi-compact, and
+the collection of quasi-compact opens forms a basis for the topology.
+
+## Main Results
+
+- `SpectralSpace` : Predicate for a topological space to be spectral.
+- `compact_openEmbedding_of_SpectralSpace_is_SpectralSpace` : proof that a compact open subspace 
+    of a spectral space is spectral
+
+## Notation
+
+- None so far.
+
+## References
+See the [StacksProject], tag 08YF for details.
+
+-/
+
+variable {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y] {f : X → Y}
+
+open TopologicalSpace
+
+/-- Let X be an open subspace of a topological space Y. If the quasi-compact opens
+of Y form a basis of Y, then the quasi-compact opens of X form a basis of X.-/
+theorem OpenEmbedding.has_qcOpenBasis_of_qcOpenBasis (hf : OpenEmbedding f)
+    (hYb : IsTopologicalBasis {S : Set Y | IsOpen S ∧ IsCompact S}) :
+    IsTopologicalBasis {S : Set X | IsOpen S ∧ IsCompact S} := by
+  apply isTopologicalBasis_of_isOpen_of_nhds (fun U hU ↦ hU.1) <| fun x S hx hS ↦ ?_
+  obtain ⟨W, ⟨hoW, hcW⟩, ⟨hfx, hWf⟩⟩ : ∃ t ∈ {S | IsOpen S ∧ IsCompact S}, f x ∈ t ∧ t ⊆ f '' S :=
+    hYb.isOpen_iff.mp (hf.open_iff_image_open.mp hS) (f x) ⟨x, hx, rfl⟩
+  refine ⟨f ⁻¹' W, ⟨hoW.preimage hf.continuous, ?_⟩, ⟨hfx, fun y hy ↦ ?_⟩⟩
+  · exact hf.toInducing.isCompact_preimage' hcW <| Set.SurjOn.subset_range hWf
+  · exact hf.inj.mem_set_image.mp (hWf hy)
```

Comment:
Can you move this to an earlier file?

Reply:
You can use `#find_home! OpenEmbedding.has_qcOpenBasis_of_qcOpenBasis` to find a good home for it.

Reply:
Also, I think the name should be changed to `OpenEmbedding.quasiCompactOpenBasis` (or else `OpenEmbedding.qcOpenBasis`, but _only if_ we are already using that abbreviation elsewhere).